src/qibolab/result.py

```diff
@@ -30,24 +40,30 @@ def i(self):
     def q(self):
         return self.array.q
 
-    @property
-    def shots(self):
-        return self.array.shots
+    def __add__(self, data):
+        i = np.append(self.i, data.i)
+        q = np.append(self.q, data.q)
```

Comment:
In case you are sure about using `np.append`, prefer:
```suggestion
        i = np.append(self.i, data.i, axis=0)
        q = np.append(self.q, data.q, axis=0)
```

It is concatenating anyhow on the most external dimension, but without `axis` parameter is flattening the array first, see [`np.append`](https://numpy.org/doc/stable/reference/generated/numpy.append.html#numpy-append) docs.
In this way, you will impose a compatible shape of the components, and won't lose the shape information in the result.